src/html5/parser.rs

```diff
@@ -1975,11 +1979,7 @@ impl<'stream> Html5Parser<'stream> {
                 // ignore token
             }
             Token::TextToken { .. } if self.current_token.is_empty_or_white() => {
-                self.reconstruct_formatting();
-
-                let node = self.create_node(&self.current_token, HTML_NAMESPACE);
-                let parent_node = current_node!(self);
-                self.document.get_mut().add_node(node, parent_node.id, None);
+                self.create_or_merge_text(self.current_token.clone());
```

Comment:
This change helps to get one of the unit tests below to pass.

Reply:
This makes sort of sense, but is not right. The add_node uses parent_id, which may or may not be the current_token.  The "Create_or_merge" basically checks if the last token is a text token, and merges any text into it.

So it's an improvement on the merging text part.. but it needs to have parent_node.id as an override of some sorts